tests/test_performance.py

```diff
@@ -0,0 +1,239 @@
+from datetime import date, timedelta
+
+import numpy as np
+import pandas as pd
+import patsy
+import pymc3 as pm
+import pytest
+import theano.tensor as tt
+
+from pymc3_hmm.distributions import DiscreteMarkovChain, SwitchingProcess
+from pymc3_hmm.step_methods import FFBSStep
+from pymc3_hmm.utils import multilogit_inv
+
+
+def gen_toy_data(days=-7 * 40):
+    dates = date.today() + timedelta(days=days), date.today() + timedelta(days=2)
+    date_rng = pd.date_range(start=min(dates), end=max(dates), freq="H")
+    raw_data = {"dt": date_rng}
+    df = pd.DataFrame.from_dict(raw_data)
+    df = df.assign(
+        month=pd.Categorical(df.dt.dt.month, categories=list(range(1, 13))),
+        weekday=pd.Categorical(df.dt.dt.weekday, categories=list(range(7))),
+        hour=pd.Categorical(df.dt.dt.hour, categories=list(range(24))),
+    )
+    return df
+
+
+def create_poisson_zero_hmm(X, mu, xis, observed):
+    S = 2
+    z_tt = tt.stack([tt.dot(X, xis[..., s, :]) for s in range(S)], axis=1)
+    Gammas_tt = pm.Deterministic("Gamma", multilogit_inv(z_tt))
+    gamma_0_rv = pm.Dirichlet("gamma_0", np.ones((S,)))
+    V_rv = DiscreteMarkovChain("V_t", Gammas_tt, gamma_0_rv, shape=X.shape[0])
+    V_rv.tag.test_value = (observed > 0) * 1
+    Y_rv = SwitchingProcess(
+        "Y_t",
+        [pm.Constant.dist(0), pm.Poisson.dist(mu)],
+        V_rv,
+        observed=observed,
+    )
+    return Y_rv
+
+
+def create_NB_zero_hmm(X, mu, xis, alpha, observed):
+    S = 2
+    z_tt = tt.stack([tt.dot(X, xis[..., s, :]) for s in range(S)], axis=1)
+    Gammas_tt = pm.Deterministic("Gamma", multilogit_inv(z_tt))
+    gamma_0_rv = pm.Dirichlet("gamma_0", np.ones((S,)))
+    V_rv = DiscreteMarkovChain("V_t", Gammas_tt, gamma_0_rv, shape=X.shape[0])
+    V_rv.tag.test_value = (observed > 0) * 1
+    Y_rv = SwitchingProcess(
+        "Y_t",
+        [pm.Constant.dist(0), pm.NegativeBinomial.dist(mu, alpha)],
+        V_rv,
+        observed=observed,
+    )
+    return Y_rv
+
+
+def assert_performance(model, y_t):
+    number_of_draws = 50
+    with model:
+        steps = [
+            FFBSStep([model.V_t]),
+            pm.NUTS(
+                vars=[
+                    model.gamma_0,
+                    model.Gamma,
+                ],
+                target_accept=0.90,
+            ),
+        ]
+
+    with model:
+        posterior_trace = pm.sample(
+            draws=number_of_draws,
+            step=steps,
+            random_seed=100,
+            return_inferencedata=True,
+            chains=1,
+            cores=1,
+            tune=number_of_draws // 2,
+            progressbar=True,
+            idata_kwargs={"dims": {"Y_t": ["dt"]}},
+        )
+
+    adds_pois_ppc = pm.sample_posterior_predictive(
+        posterior_trace.posterior, var_names=["Y_t"], model=model
+    )
+
+    assert (adds_pois_ppc["Y_t"].mean(0) - y_t).std() < y_t.std() / 10
```

Comment:
We should use a more "all-purpose" check here, because the relationship between a measure like this and the std. dev. of the observations will be affected by too many factors that change between tests (e.g. the changes between models, parameters, etc.)

For instance, we could check whether or not the true/observed values lie within a density/credible interval (e.g. a highest density interval from `az.hdi`) of the posterior/posterior predictive samples.

The interval approach has its shortcomings (e.g. when faced with sample results containing well separated mixtures), but it using it in/for these cases should be fine.